### ISSUE TITLE
Dependency dependency (transitive?)

### ISSUE THREAD

**Issue by sbv-siteimprove** [kardianos/govendor#348]
Hey,

When I added sirupsen/logrus as a dependency to my project this ended up in the vendor.json:

````
		{
			"checksumSHA1": "2ncnCWXu6MhDubZJYmyFNBYsrXE=",
			"origin": "github.com/sirupsen/logrus/vendor/golang.org/x/sys/unix",
			"path": "golang.org/x/sys/unix",
			"revision": "94b76065f2d2081d0fef24a6e67c571f51a6408a",
			"revisionTime": "2017-07-01T06:49:51Z"
		},
``` 

but when doing `govendor sync` on a clean checkout that  package always fails with
```` 
fatal: Could not parse object '94b76065f2d2081d0fef24a6e67c571f51a6408a'
````
in 
``` .cache/govendor/github.com/sirupsen/logrus/; git reset --hard 94b76065f2d2081d0fef24a6e67c571f51a6408a
```` 
and surely enough sirupsen/logrus dont have any commits with that revision. If I update logrus to a newer revision this revision gets updated as well but the problem persists.

I'm guessing that the revision is a revision within x/sys/unix and not logrus?

We removed those lines from vendor.json and added x/sys/unix as a toplevel dependency instead and that seems to solve the problem - I'm just curious if we did something wrong in the first go-around?

**Comment by AlekSi** [comment]
I have the same problem with govendor in Prometheus.

**Comment by AlekSi** [comment]
> I'm guessing that the revision is a revision within x/sys/unix and not logrus?

Yes: https://github.com/golang/sys/commit/94b76065f2d2081d0fef24a6e67c571f51a6408a

**Comment by kaey** [comment]
Can you make a small repro?
I'm intrested in how you end up with ```github.com/sirupsen/logrus/vendor/golang.org/x/sys/unix``` in your origin.

**Comment by AlekSi** [comment]
Related to https://github.com/prometheus/prometheus/issues/3030. Not sure what happened there yet.
